### PR TITLE
fix TestASM test flakes

### PR DESCRIPTION
**Pull request by freehan**
cc @skmatti @spencerhance

### CODE REVIEW

File: cmd/e2e-test/asm_test.go [hunk 1]
```diff
@@ -162,8 +162,7 @@ func TestASMServiceAndDestinationRule(t *testing.T) {
 					}
 				} else {
 					if negName, ok := negStatus.NetworkEndpointGroups[strconv.Itoa(int(porterPort))]; ok {
-						// No backend pod exists, so the NEG has 0 endpoint.
-						if err := e2e.WaitForNegs(ctx, Framework.Cloud, negName, negStatus.Zones, false, 0); err != nil {
+						if err := e2e.WaitForNegs(ctx, Framework.Cloud, negName, negStatus.Zones, false, 6); err != nil {
 							t.Errorf("Failed to wait Negs, error: %s", err)
```
Comment (prameshj) [inline cmd/e2e-test/asm_test.go:166]: Can we derive this from number of NEG ports configured? or some other const?

Comment (freehan) [reply cmd/e2e-test/asm_test.go:166]: oh. this is the total number of replicas in the above deployments

Comment (freehan) [reply cmd/e2e-test/asm_test.go:166]: oh sorry. i take that back

**Pull request merged**